Assets/Scripts/UI/InteractTrigger.cs

```diff
@@ -0,0 +1,36 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class InteractTrigger : MonoBehaviour {
```

Comment:
are you going to attach this to all the objects? wouldn't it be nicer if you create a separate collider for the player doll, and then check if the game object that comes within range can be interacted with, then show the UI?